src/main/java/seedu/address/logic/commands/ListCommand.java

```diff
@@ -2,20 +2,45 @@
 
 import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
 
+import java.util.function.Predicate;
+
+import seedu.address.model.person.ReadOnlyPerson;
+
 /**
  * Lists all persons in the address book to the user.
  */
 public class ListCommand extends Command {
 
     public static final String COMMAND_WORD = "list";
     public static final String COMMAND_ALIAS = "l";
+    public static final String COMMAND_OPTION_FAV = "-" + FavoriteCommand.COMMAND_WORD;
+
+    public static final String MESSAGE_SUCCESS_LIST_ALL = "Listed all persons";
+    public static final String MESSAGE_SUCCESS_LIST_FAV = "Listed all favorite persons";
 
-    public static final String MESSAGE_SUCCESS = "Listed all persons";
+    private boolean hasOptionFav = false;
 
+    public ListCommand(String args) {
+        if (args.trim().equals(COMMAND_OPTION_FAV)) {
+            hasOptionFav = true;
+        }
+    }
 
     @Override
     public CommandResult execute() {
-        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
-        return new CommandResult(MESSAGE_SUCCESS);
+        if (hasOptionFav) {
+            model.updateFilteredPersonList(showAllFavoritePersons());
+            return new CommandResult(MESSAGE_SUCCESS_LIST_FAV);
+        } else {
+            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+            return new CommandResult(MESSAGE_SUCCESS_LIST_ALL);
+        }
+    }
+
+    /**
+     * Returns a predicate consisting of all ReadOnlyPerson who has been favorited
+     */
+    public static Predicate<ReadOnlyPerson> showAllFavoritePersons() {
```

Comment:
Should we put this in a separate class, similar to how PREDICATE_SHOW_ALL_PERSONS?

Reply:
Good point! Updated ✔